<a href="https://colab.research.google.com/github/yasir-ansari951/flyrank-ml-internship/blob/main/work/notebooks/copy_of_w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# My Lane

I selected the Content Refresh Prioritization lane.

This lane focuses on identifying pages that are most likely to benefit from content updates. It can help SEO teams prioritize their limited resources toward pages with the highest potential impact.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

# Research Question

Question:
Which pages are most likely to benefit from a content refresh?

Decision:
Which pages should be updated first?

Action:
Refresh the selected pages.

Cost of a wrong recommendation:
The team may spend time updating pages that do not improve performance while missing pages with greater potential.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import os, subprocess

if not os.path.isdir("flyrank-ml-internship-starter"):
    subprocess.run([
        "git", "clone", "--depth", "1",
        "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
    ])

os.chdir("flyrank-ml-internship-starter")

In [ ]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")
print(df.head())

             content_id          client_id  search_volume  competition  \
0  content_304f48230142  client_f369cb89fc           10.0         0.67   
1  content_a1fb4e703a9e  client_4e07408562           90.0         0.01   
2  content_9aa793d4d895  client_7f2253d7e2            0.0         0.00   
3  content_331d6c4de07b  client_19581e27de           10.0         0.00   
4  content_d99b7a2d90ca  client_3fdba35f04            0.0         0.00   

  competition_level   cpc     content_type    main_intent  word_count  \
0              HIGH  2.05  keyword article  transactional      3221.0   
1               LOW  0.05  keyword article  informational      2481.0   
2               LOW  0.00  keyword article  informational      3515.0   
3               LOW  0.00  keyword article     commercial         NaN   
4               LOW  0.00  keyword article  informational      2803.0   

   char_count  ... char_count_tier   ctr  avg_position  engagement_rate  \
0     20457.0  ...     15000-25000  0.76 

In [ ]:
print("Rows:", len(df))
print("Columns:", len(df.columns))
print("Average CTR:", round(df["ctr"].mean(), 3))
print("Median Impressions:", df["impressions_90d"].median())
print("Average Position:", round(df["avg_position"].mean(), 2))

Rows: 30000
Columns: 44
Average CTR: 0.511
Median Impressions: 731.0
Average Position: 16.34


In [ ]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))
print("Average CTR:", round(df["ctr"].mean(), 3))
print("Median Impressions:", df["impressions_90d"].median())
print("Average Position:", round(df["avg_position"].mean(), 2))

Rows: 30000
Columns: 44
Average CTR: 0.511
Median Impressions: 731.0
Average Position: 16.34


# Quick Look at the Data

The starter dataset contains 30,000 pages and 44 variables.

The average CTR is 0.511.

The median number of impressions is 731.

The average position is 16.34.

These numbers suggest that page performance varies across the dataset, making prioritization a potentially useful decision-support problem.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

# Careful Words

This project will look for patterns in the available data.

Any findings will be observational and should not be interpreted as proof of causation.

The goal is to support decision-making rather than make certain predictions about future outcomes.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.

# Self Check

✓ Selected a lane

✓ Defined the question

✓ Defined the decision

✓ Defined the action

✓ Defined the cost of a wrong recommendation

✓ Included supporting data

✓ Used careful language